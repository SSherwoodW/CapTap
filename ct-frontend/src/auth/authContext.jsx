import { createContext, useContext, useState, useEffect } from 'react';
import useLocalStorage from '../hooks/useLocalStorage';
import CapTapApi from '../../../api';
import * as jose from 'jose';


export const AuthContext = createContext();

export const useAuth = () => useContext(AuthContext);

export const AuthProvider = ({ children }) => {
    const [isLoggedIn, setIsLoggedIn] = useState();
    const [currentUser, setCurrentUser] = useLocalStorage("currentUser", null);
    const [token, setToken] = useLocalStorage("token", null);

    useEffect(() => {
        async function fetchCurrentUser() {
            try {
                if (token !== null) {
                    CapTapApi.token = token;
                    const { username } = jose.decodeJwt(token);
                    const currentUser = await CapTapApi.getCurrentUser(username);
                    setCurrentUser(currentUser);
                    setIsLoggedIn(true);
                }
            } catch (error) {
                console.error('Error fetching current user:', error);
                setCurrentUser(null);
                setIsLoggedIn(false);
            }
        }

        fetchCurrentUser();
    }, [token]);

    async function handleLogin(formData) {
        try {
            const logInResponse = await CapTapApi.login(formData);
            setToken(logInResponse);
            setIsLoggedIn(true);

            // Fetch user data and set currentUser state
            const { username } = jose.decodeJwt(token);
            const userData = await CapTapApi.getCurrentUser(username);
            setCurrentUser(userData);
            return { success: true };
        } catch (err) {
            console.log("login failed", err)
            return {success: false, error: err.message};
        }
    };

    async function handleSignup(formData) {
        try {
            const signUpResponse = await CapTapApi.signup(formData);
            setToken(signUpResponse);
            localStorage.setItem('token', signUpResponse);
            setIsLoggedIn(true);

            // Fetch user data and set currentUser state
            const { username } = jose.decodeJwt(token);
            const userData = await CapTapApi.getCurrentUser(username);
            setCurrentUser(userData);
            return { success: true };
        } catch (err) {
            console.error("signup failed", err);
            return { success: false, error: err.message };
        }
    };

    const handleLogout = () => {
        localStorage.removeItem('token');
        setToken(null);
        setIsLoggedIn(false);
        setCurrentUser(null);
    };

    return (
        <AuthContext.Provider value={{ isLoggedIn, currentUser, handleLogin, handleSignup, handleLogout }}>
            {children}
        </AuthContext.Provider>
    );
}

export default AuthProvider;