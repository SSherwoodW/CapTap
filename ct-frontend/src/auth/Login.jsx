import React, { useContext, useState } from "react";
import { Navigate } from "react-router-dom";
import { AuthContext } from "./authContext";


function Login() {
    const { handleLogin, currentUser } = useContext(AuthContext);
    const [formData, setFormData] = useState({
        username: "",
        password: "",
    });
    const [formErrors, setFormErrors] = useState([]);

    console.debug(
        "LoginForm",
        "login=", typeof handleLogin,
        "formData=", formData,
        "formErrors", formErrors,
    );

    /** Handle form submit:
     *
     * Calls login func prop and, if successful, redirect to /.
     */

    async function handleSubmit(evt) {
        evt.preventDefault();
        console.log("formData in handleSubmit", formData);
        let result = await handleLogin(formData);
        if (result.success) {
        console.log('success in handleSubmit Login')
        } else {
        setFormErrors(result.errors);
        }
    };

    const handleChange = (evt) => {
        const { name, value } = evt.target;
        setFormData(data => ({
            ...data,
            [name]: value
        }));
    };


    return (
        <div className="flex min-h-full flex-col justify-center px-6 py-12 lg:px-8">
            {currentUser && (
                <Navigate to="/" replace={true} />
            )}
            <div className="sm:mx-auto sm:w-full sm:max-w-sm">
                <img className="mx-auto h-10 w-auto" src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=600" alt="CapTap"></img>
                <h2 className="mt-10 text-center text-2xl font-bold leading-9 tracking-tight text-gray-900">Sign in to your account</h2>
            </div>

            <div className="mt-10 sm:mx-auto sm:w-full sm:max-w-sm">
                <form className="space-y-6" onSubmit={handleSubmit}>
                    <div>
                        <label htmlFor="username" className="block text-sm font-medium leading-6 text-gray-900">Username</label>
                        <div className="mt-2">
                            <input id="username" name="username" type="text" value={formData.username} onChange={handleChange} required className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"/>
                        </div>
                    </div>

                    <div>
                        <div className="flex items-center justify-between">
                            <label htmlFor="password" className="block text-sm font-medium leading-6 text-gray-900">Password</label>
                            <div className="text-sm">
                                <a href="#" className="font-semibold text-indigo-600 hover:text-indigo-500">Forgot password?</a>
                            </div>
                        </div>
                        <div className="mt-2">
                            <input id="password" name="password" type="password" value={formData.password} onChange={handleChange} required className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"/>
                        </div>
                    </div>

                    <div>
                        <button type="submit" className="flex w-full justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600">Sign in</button>
                    </div>
                </form>

                <p className="mt-10 text-center text-sm text-gray-500">
                Not a member?
                    <a href="#" className="font-semibold leading-6 text-indigo-600 hover:text-indigo-500">Sign Up!</a>
                </p>
            </div>
        </div>
    )
}

export default Login;