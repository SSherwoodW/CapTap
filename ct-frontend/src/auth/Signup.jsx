import React, { useContext, useState } from "react";
import { Navigate } from "react-router-dom";
import { AuthContext } from "./authContext";

function Signup() {
    const { currentUser, handleSignup } = useContext(AuthContext);
    const [formData, setFormData] = useState({
        firstName: "",
        lastName: "",
        email: "",
        username: "",
        birthdate: "",
        password: "",
    });
    const [formErrors, setFormErrors] = useState([]);

    console.debug(
        "SignupForm",
        "signup=", typeof handleSignup,
        "formData=", formData,
        "formErrors", formErrors,
    );

    /** Handle form submit:
     *
     * Calls signup func prop and, if successful, redirect to /.
     */

    async function handleSubmit(evt) {
        evt.preventDefault();
        console.log("formData in handleSubmit", formData);
        let result = await handleSignup(formData);
        if (result.success) {
        console.log('success in handleSubmit Signup')
        } else {
        setFormErrors(result.errors);
        }
    };

    const handleChange = (evt) => {
        const { name, value } = evt.target;
        setFormData(data => ({
            ...data,
            [name]: value
        }));
    };
    return <>
        <div className="flex min-h-full flex-col justify-center px-6 py-12 lg:px-8">
            {currentUser && (
                <Navigate to="/" replace={true} />
            )}
            <div className="sm:mx-auto sm:w-full sm:max-w-sm">
                <img className="mx-auto h-10 w-auto" src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=600" alt="CapTap"></img>
                <h2 className="mt-10 text-center text-2xl font-bold leading-9 tracking-tight text-gray-900">Sign up for your account</h2>
            </div>

            <div className="mt-10 sm:mx-auto sm:w-full sm:max-w-sm">
                <form className="space-y-4" onSubmit={handleSubmit}>
                    <div className="flex items-center justify-between">
                        <div>
                            <label htmlFor="firstName" className="block text-sm font-medium leading-6 text-gray-900">First Name</label>
                            <div className="mt-2">
                                <input id="firstName" name="firstName" type="text" value={formData.firstName} onChange={handleChange} required className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"/>
                            </div>
                        </div>
                        <div>
                            <label htmlFor="lastName" className="block text-sm font-medium leading-6 text-gray-900">Last Name</label>
                            <div className="mt-2">
                                <input id="lastName" name="lastName" type="text" value={formData.lastName} onChange={handleChange} required className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"/>
                            </div>
                        </div>
                    </div>
                    <div>
                        <label htmlFor="birthdate" className="block text-sm font-medium leading-6 text-gray-900">Birthdate</label>
                        <div className="mt-2">
                            <input id="birthdate" name="birthdate" type="date" value={formData.birthdate} onChange={handleChange} required className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"/>
                        </div>
                    </div>
                    <div>
                        <label htmlFor="email" className="block text-sm font-medium leading-6 text-gray-900">Email</label>
                        <div className="mt-2">
                            <input id="email" placeholder="ex: you@captap.com" name="email" type="email" value={formData.email} onChange={handleChange} required className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"/>
                        </div>
                    </div>
                    <div>
                        <label htmlFor="username" className="block text-sm font-medium leading-6 text-gray-900">Username</label>
                        <div className="mt-2">
                            <input id="username" autoComplete="username" name="username" type="text" value={formData.username} onChange={handleChange} required className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"/>
                        </div>
                    </div>

                    <div>
                        <div className="flex items-center justify-between">
                            <label htmlFor="password" className="block text-sm font-medium leading-6 text-gray-900">Password</label>
                            <div className="text-sm">
                                <a href="#" className="font-semibold text-indigo-600 hover:text-indigo-500">Forgot password?</a>
                            </div>
                        </div>
                        <div className="mt-2">
                            <input id="password" autoComplete="current-password" name="password" type="password" value={formData.password} onChange={handleChange} required className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"/>
                        </div>
                    </div>

                    <div>
                        <button type="submit" className="flex w-full justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600">Sign up</button>
                    </div>
                </form>

                <p className="mt-10 text-center text-sm text-gray-500">
                Already registered?
                    <a href="/login" className="font-semibold leading-6 text-indigo-600 hover:text-indigo-500"> Log In</a>
                </p>
            </div>
        </div>
    </>
}

export default Signup;