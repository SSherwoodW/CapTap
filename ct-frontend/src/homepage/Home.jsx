import { useContext } from "react";
import { Link, Outlet, useLocation } from "react-router-dom";

import Navigation from "../routes-nav/Navigation";
import { AuthContext } from "../auth/authContext";


function Home() {
    const { currentUser } = useContext(AuthContext);

    const location = useLocation();
    const isHomePage = location.pathname === '/';

    return (
        <div>
            <Navigation />
            {isHomePage && (
                <div className="">
                    <p>Home Page Content</p>
                    <Link to="/parlayer">Go to Other Page</Link>
                </div>
            )}
            <div id="detail">
                <Outlet />
            </div>
        </div>
    );
}

export default Home;