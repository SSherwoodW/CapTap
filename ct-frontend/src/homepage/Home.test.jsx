import React from "react";
import { render, screen } from '@testing-library/react';
import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
import { AuthContext } from '../auth/authContext'; 

import Home from './Home';

vi.mock('react-router-dom', async () => {
    const mod = await vi.importActual('react-router-dom');
    return {
        ...mod,
        useLocation: () => ({
            location: {pathname: "/"}
        })
    }
})

describe('Home', () => {
    beforeEach(() => {
        const mockAuthContext = {
        currentUser: { username: 'testuser' } 
        };
        vi.spyOn(React, 'useContext').mockReturnValue(mockAuthContext);
    });

    afterEach(() => {
        vi.restoreAllMocks();
    });
    
    it('renders the Home component', () => {
        render(
            <AuthContext.Provider value={{ currentUser: { username: 'testuser' } }}>
                <Home />
            </AuthContext.Provider>
        )
        screen.debug();
    });

    it('matches snapshot', () => {
        const homePage = render(
            <AuthContext.Provider value={{ currentUser: { username: 'testuser' } }}>
                <Home />
            </AuthContext.Provider>
        )
        screen.debug();
        expect(homePage).toMatchSnapshot();
    });

    it('matches snapshot when logged out', () => {
        const homePage = render(
            <AuthContext.Provider value={{ currentUser: null }}>
                <Home />
            </AuthContext.Provider>
        )
        screen.debug();
        expect(homePage).toMatchSnapshot();
    });
})