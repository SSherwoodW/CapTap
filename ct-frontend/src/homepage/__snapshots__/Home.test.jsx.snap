// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="bg-gray-800"
          data-headlessui-state=""
        >
          <div
            class="mx-auto max-w-8xl px-1 sm:px-6 lg:px-8"
          >
            <div
              class="relative flex h-16 items-center justify-between"
            >
              <div
                class="absolute inset-y-0 left-0 flex items-center sm:hidden"
              >
                <button
                  aria-expanded="false"
                  class="relative inline-flex items-center justify-center rounded-md p-2 text-gray-400 hover:bg-gray-700 hover:text-white focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white"
                  data-headlessui-state=""
                  id="headlessui-disclosure-button-:r3:"
                  type="button"
                >
                  <span
                    class="absolute -inset-0.5"
                  />
                  <span
                    class="sr-only"
                  >
                    Open main menu
                  </span>
                  <svg
                    aria-hidden="true"
                    class="block h-6 w-6"
                    data-slot="icon"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="flex flex-1 items-center justify-center sm:items-stretch sm:justify-start"
              >
                <div
                  class="flex flex-shrink-0 items-center"
                >
                  <img
                    alt="CapTap Logo"
                    class="h-8 w-auto"
                    src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=500"
                  />
                </div>
                <div
                  class="hidden sm:ml-6 sm:flex"
                >
                  <div
                    class="flex items-center justify-center sm:items-stretch sm:justify-start"
                  >
                    <a
                      aria-current="page"
                      class="bg-gray-900 text-white rounded-md px-3 py-2 text-sm font-medium"
                      href="/"
                    >
                      Home
                    </a>
                    <a
                      class="text-gray-300 hover:bg-gray-700 hover:text-white rounded-md px-3 py-2 text-sm font-medium"
                      href="/parlayer"
                    >
                      Parlayer
                    </a>
                    <a
                      class="text-gray-300 hover:bg-gray-700 hover:text-white rounded-md px-3 py-2 text-sm font-medium"
                      href="/journal/testuser"
                    >
                      Journal
                    </a>
                    <div>
                      <button
                        class="text-gray-300 hover:bg-gray-700 hover:text-white rounded-md px-3 py-2 text-sm font-medium"
                      >
                        Search Teams
                      </button>
                    </div>
                    <div>
                      <button
                        class="text-gray-300 hover:bg-gray-700 hover:text-white rounded-md px-3 py-2 text-sm font-medium"
                      >
                        Search Players
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0"
              >
                <div
                  class="relative ml-3"
                  data-headlessui-state=""
                >
                  <div>
                    <button
                      aria-expanded="false"
                      aria-haspopup="menu"
                      class="relative flex rounded-full bg-gray-800 text-sm focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-gray-800"
                      data-headlessui-state=""
                      id="headlessui-menu-button-:r4:"
                      type="button"
                    >
                      <span
                        class="absolute -inset-1.5"
                      />
                      <span
                        class="sr-only"
                      >
                        Open user menu
                      </span>
                      <img
                        alt="user profile photo"
                        class="h-8 w-8 rounded-full"
                        src=""
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </nav>
        <div
          id="detail"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="bg-gray-800"
        data-headlessui-state=""
      >
        <div
          class="mx-auto max-w-8xl px-1 sm:px-6 lg:px-8"
        >
          <div
            class="relative flex h-16 items-center justify-between"
          >
            <div
              class="absolute inset-y-0 left-0 flex items-center sm:hidden"
            >
              <button
                aria-expanded="false"
                class="relative inline-flex items-center justify-center rounded-md p-2 text-gray-400 hover:bg-gray-700 hover:text-white focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white"
                data-headlessui-state=""
                id="headlessui-disclosure-button-:r3:"
                type="button"
              >
                <span
                  class="absolute -inset-0.5"
                />
                <span
                  class="sr-only"
                >
                  Open main menu
                </span>
                <svg
                  aria-hidden="true"
                  class="block h-6 w-6"
                  data-slot="icon"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
            <div
              class="flex flex-1 items-center justify-center sm:items-stretch sm:justify-start"
            >
              <div
                class="flex flex-shrink-0 items-center"
              >
                <img
                  alt="CapTap Logo"
                  class="h-8 w-auto"
                  src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=500"
                />
              </div>
              <div
                class="hidden sm:ml-6 sm:flex"
              >
                <div
                  class="flex items-center justify-center sm:items-stretch sm:justify-start"
                >
                  <a
                    aria-current="page"
                    class="bg-gray-900 text-white rounded-md px-3 py-2 text-sm font-medium"
                    href="/"
                  >
                    Home
                  </a>
                  <a
                    class="text-gray-300 hover:bg-gray-700 hover:text-white rounded-md px-3 py-2 text-sm font-medium"
                    href="/parlayer"
                  >
                    Parlayer
                  </a>
                  <a
                    class="text-gray-300 hover:bg-gray-700 hover:text-white rounded-md px-3 py-2 text-sm font-medium"
                    href="/journal/testuser"
                  >
                    Journal
                  </a>
                  <div>
                    <button
                      class="text-gray-300 hover:bg-gray-700 hover:text-white rounded-md px-3 py-2 text-sm font-medium"
                    >
                      Search Teams
                    </button>
                  </div>
                  <div>
                    <button
                      class="text-gray-300 hover:bg-gray-700 hover:text-white rounded-md px-3 py-2 text-sm font-medium"
                    >
                      Search Players
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0"
            >
              <div
                class="relative ml-3"
                data-headlessui-state=""
              >
                <div>
                  <button
                    aria-expanded="false"
                    aria-haspopup="menu"
                    class="relative flex rounded-full bg-gray-800 text-sm focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-gray-800"
                    data-headlessui-state=""
                    id="headlessui-menu-button-:r4:"
                    type="button"
                  >
                    <span
                      class="absolute -inset-1.5"
                    />
                    <span
                      class="sr-only"
                    >
                      Open user menu
                    </span>
                    <img
                      alt="user profile photo"
                      class="h-8 w-8 rounded-full"
                      src=""
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </nav>
      <div
        id="detail"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home > matches snapshot when logged out 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="bg-gray-800"
          data-headlessui-state=""
        >
          <div
            class="mx-auto max-w-8xl px-1 sm:px-6 lg:px-8"
          >
            <div
              class="relative flex h-16 items-center justify-between"
            >
              <div
                class="absolute inset-y-0 left-0 flex items-center sm:hidden"
              >
                <button
                  aria-expanded="false"
                  class="relative inline-flex items-center justify-center rounded-md p-2 text-gray-400 hover:bg-gray-700 hover:text-white focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white"
                  data-headlessui-state=""
                  id="headlessui-disclosure-button-:r6:"
                  type="button"
                >
                  <span
                    class="absolute -inset-0.5"
                  />
                  <span
                    class="sr-only"
                  >
                    Open main menu
                  </span>
                  <svg
                    aria-hidden="true"
                    class="block h-6 w-6"
                    data-slot="icon"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="flex flex-1 items-center justify-center sm:items-stretch sm:justify-start"
              >
                <div
                  class="flex flex-shrink-0 items-center"
                >
                  <img
                    alt="CapTap Logo"
                    class="h-8 w-auto"
                    src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=500"
                  />
                </div>
                <div
                  class="hidden sm:ml-6 sm:block"
                >
                  <div
                    class="flex space-x-4"
                  >
                    <a
                      aria-current="page"
                      class="bg-gray-900 text-white rounded-md px-3 py-2 text-sm font-medium"
                      href="/"
                    >
                      Home
                    </a>
                    <a
                      class="text-gray-300 hover:bg-gray-700 hover:text-white rounded-md px-3 py-2 text-sm font-medium"
                      href="/login"
                    >
                      Login
                    </a>
                    <a
                      class="text-gray-300 hover:bg-gray-700 hover:text-white rounded-md px-3 py-2 text-sm font-medium"
                      href="/signup"
                    >
                      Signup
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </nav>
        <div
          id="detail"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="bg-gray-800"
        data-headlessui-state=""
      >
        <div
          class="mx-auto max-w-8xl px-1 sm:px-6 lg:px-8"
        >
          <div
            class="relative flex h-16 items-center justify-between"
          >
            <div
              class="absolute inset-y-0 left-0 flex items-center sm:hidden"
            >
              <button
                aria-expanded="false"
                class="relative inline-flex items-center justify-center rounded-md p-2 text-gray-400 hover:bg-gray-700 hover:text-white focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white"
                data-headlessui-state=""
                id="headlessui-disclosure-button-:r6:"
                type="button"
              >
                <span
                  class="absolute -inset-0.5"
                />
                <span
                  class="sr-only"
                >
                  Open main menu
                </span>
                <svg
                  aria-hidden="true"
                  class="block h-6 w-6"
                  data-slot="icon"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
            <div
              class="flex flex-1 items-center justify-center sm:items-stretch sm:justify-start"
            >
              <div
                class="flex flex-shrink-0 items-center"
              >
                <img
                  alt="CapTap Logo"
                  class="h-8 w-auto"
                  src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=500"
                />
              </div>
              <div
                class="hidden sm:ml-6 sm:block"
              >
                <div
                  class="flex space-x-4"
                >
                  <a
                    aria-current="page"
                    class="bg-gray-900 text-white rounded-md px-3 py-2 text-sm font-medium"
                    href="/"
                  >
                    Home
                  </a>
                  <a
                    class="text-gray-300 hover:bg-gray-700 hover:text-white rounded-md px-3 py-2 text-sm font-medium"
                    href="/login"
                  >
                    Login
                  </a>
                  <a
                    class="text-gray-300 hover:bg-gray-700 hover:text-white rounded-md px-3 py-2 text-sm font-medium"
                    href="/signup"
                  >
                    Signup
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </nav>
      <div
        id="detail"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
